### DISCUSSION TITLE
Different output values for backtracking

### DISCUSSION THREAD

**Discussion by pohzim** [WAM2layers/WAM2layers#407] [Q&A]
In my current workflow, I need to backtrack single days for several days. This led me to a doubt about the total values of ['tagged_precip'] for a given period, so I compared the outputs from the sum of single experiments and one experiment covering the whole period. I used the same input for both experiments, only changing the period of analysis.

The whole period (ds_wh) covers:
```
output_folder: ./2023-03w
tracking_start_date: "2023-03-01T00:00"
tracking_end_date: "2023-03-31T23:00"
tagging_start_date: "2023-03-29T00:00"
tagging_end_date: "2023-03-31T23:00"

```

For the separated experiment, I runned WAM2layers 3 times, outputing each run in a different folder. The configuration files are as follows:

ds_d1:
```
output_folder: ./2023-03-31
tracking_start_date: "2023-03-01T00:00"
tracking_end_date: "2023-03-31T23:00"
tagging_start_date: "2023-03-31T00:00"
tagging_end_date: "2023-03-31T23:00"
```
ds_d2:

```
output_folder: ./2023-03-30
tracking_start_date: "2023-03-01T00:00"
tracking_end_date: "2023-03-30T23:00"
tagging_start_date: "2023-03-30T00:00"
tagging_end_date: "2023-03-30T23:00"
```
ds_d3:

```
output_folder: ./2023-03-29
tracking_start_date: "2023-03-01T00:00"
tracking_end_date: "2023-03-29T23:00"
tagging_start_date: "2023-03-29T00:00"
tagging_end_date: "2023-03-29T23:00"
```

After the separated experiments completion, I opened all files with:

`ds_sep = xr.open_mfdataset(nc_files,concat_dim='time',combine='nested').groupby('time').sum(dim='time')`

Where `nc_files` is a list of all files paths from the separated output_folders.

To compare the outputs, I used a boolean mask:

```
mask = (ds_sep['tagged_precip'].sum(dim='time') != ds_jun['tagged_precip'].sum(dim='time')).compute()
filtered = testa.where(mask,drop=True)
```

The filtered output should be an array filled with nan, since the `tagged_precip` variable for the period is expected to be the same. However, that is not the observed behaviour. The output has lots of elements with different values. Some elements have the same value of tagged_precip. I also compared the tagging_region and the filtered output to see if it the values are different for the whole area of interest, but some elements are equal, which implies some unexpected behaviour.

```
filtered.count().values
area.where(area!=0,drop=True).count().values
```

956
1556

The max and mean values are also very different. 

ds_sep.values.max()
48509484.0
ds_sep.values.mean()
146569.27

ds_wh.values.max()
48753100.0
ds_wh.values.mean()
149829.78

I wonder if anyone has an ideia of what could cause this error. Thanks in advance!

**Comment by Peter9192** [comment]
Dear @pohzim,

That's an interesting analysis, thanks for the detailed description! I don't have time to look into it fully, but my first suspicion is that the precipitation from 23:00-00:00 might missing in the separate experiments, but included in the multi-day run. I would have a good look at the code to verify when output is written, what times are included with what timestamp assigned to them.

Cheers,
Peter
